### ISSUE TITLE
Styles (transformations) not applied, when isOpen set to true initially

### ISSUE THREAD

**Issue by hedgehog34** [negomi/react-burger-menu#70]
When initial value of `isOpen` is set to `true` on `<Menu>` component, then transformation of main content is not happening and console spits out `Element with ID 'page-wrap' not found` and `Element with ID 'outer-container' not found`.

Looks like when `handleExternalWrapper()` is being called for the first time, 'outer-container' div and 'page-wrap' are not in the DOM yet, therefore, relevant styling (transformation) is not applied to the content.

**Comment by hedgehog34** [comment]
I did manage to make this working by setting initial state to `isOpen` prop, so in `menuFactory`:
```
getDefaultProps: function getDefaultProps() {
      return {
        id: '',
        noOverlay: false,
        onStateChange: function onStateChange() {},
        outerContainerId: '',
        pageWrapId: '',
        styles: {},
        width: 300,
        breakpoint: 960,
        isOpen: false
      };
    },

getInitialState: function getInitialState() {
      return {
        isOpen: this.props.isOpen
      };
    },

componentDidMount: function componentDidMount() {
      window.onkeydown = this.listenForClose;
      if (this.props.isOpen) {
        this.toggleMenu();
      }
    },
```
Opening menu initially via `isOpen` prop:
 `<Menu isOpen ... />`

For a moment I thought this is antipattern ([Props in `getInitialState` is an Antipattern](https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html)), but clearly it is not, according to last part of this React's guide, as we are properly using `isOpen` prop to "seed data for the component's internally-controlled state".

One thing to concerning here is fact that all of this doesn't pass tests related to initial `isOpen` state:

1. menuFactory when rendered successfully is initially closed
2. menuFactory open state change should not occur when parent component state changes
3. menuFactory open state change should not occur when receiving new props if isOpen prop is undefined

**Comment by negomi** [comment]
Hi @hedgehog34, I'm glad that fixes it, but I'd need to look into this a bit more as those tests cover a lot of edge cases people have reported while using `isOpen`.

I don't have a lot of time to spend on this at the moment (sorry I haven't got back to you about your PR yet either) but will get round to it as soon as I can. Thanks so much for your detailed reporting! :)

**Comment by abologna-r7** [comment]
:1 also confirming this bug :)

**Comment by negomi** [comment]
Hey @hedgehog34 and @abologna-r7,

This should be fixed in v1.9.2.

I agree it seems not to count as an antipattern to reference the `isOpen` prop (when it exists) in this case, as it is used to seed initial data.

If you're interested, the way I solved the issue of those failing tests was to check for the presence of the `isOpen` prop inside `getInitialState`, rather than setting it as a default prop (see https://github.com/negomi/react-burger-menu/commit/5a2d98214c81589ae78b91ef090a9319abc5e349).
